src/Microsoft.Diagnostics.EventFlow.Inputs.Etw/StandardTraceEventSession.cs

```diff
@@ -0,0 +1,90 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Session;
+using Validation;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs
+{
+    internal class StandardTraceEventSession : ITraceEventSession
+    {
+        private TraceEventSession inner;
+        private bool isProcessing;
+        private IHealthReporter healthReporter;
+
+        public StandardTraceEventSession(IHealthReporter healthReporter)
+        {
+            Requires.NotNull(healthReporter, nameof(healthReporter));
+
+            this.healthReporter = healthReporter;
+            this.inner = new TraceEventSession($"EventFlow-{nameof(EtwInput)}-{Guid.NewGuid().ToString()}", TraceEventSessionOptions.Create);
+            this.inner.Source.Dynamic.All += OnEvent;
+            this.isProcessing = false;
+        }
+
+        public event Action<EventData> AllEvents;
```

Comment:
Very good suggestion. I ended up requiring a delegate to be passed to the Process() method of the ITraceEventSession interface.